### PR TITLE
Linter: Amend docs for global_bear functionality

### PR DESCRIPTION
**Pull request by NiklasMM**
Add documentation on how to use @linter to create global
bears, both in the in-line documentation and in
Writing_Linter_Bears.rst.

Fixes https://github.com/coala/coala/issues/3968

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [ ] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: docs/Developers/Writing_Linter_Bears.rst [hunk 1]
```diff
@@ -10,6 +10,12 @@ order to integrate linters in your bears.
   wrapping a tool, please refer to
   :doc:`this link instead<Writing_Native_Bears>`.
 
+  Also this tutorial takes you through the process of writing a local linter
```
Comment (sims1253) [inline docs/Developers/Writing_Linter_Bears.rst]: "This tutorial also takes you" ... I think starting with also is a german thing

Comment (NiklasMM) [inline docs/Developers/Writing_Linter_Bears.rst]: JA! AND WHATZ WRONG WIS SZATT?

**Comment by NiklasMM** [comment]
@sims1253 Improved stuff! :+1:

**Comment by sims1253** [comment]
somehow related, you could fix this while you are at it:
#3975 :D

**Comment by NiklasMM** [comment]
Sure, I'll give it a try. :+1:

**Comment by NiklasMM** [comment]
@sims1253 I added another commit fixing the issue you referenced. I kept it short, since `config_file` is probably going to be deprecated at some point. (https://github.com/coala/coala/issues/2067)

File: docs/Developers/Writing_Linter_Bears.rst [hunk 2]
```diff
@@ -71,7 +71,8 @@ appropriate parameters. This is done inside ``create_arguments``,
 - ``file``: The contents of the file to process, given as a list of lines
   (including the return character).
 - ``config_file``: The absolute path to a config file to use. If no config file
-  is used, this parameter is ``None``. More on that later.
+  is used, this parameter is ``None``. Processing of the config file
+  is left to the Bear's implementation of the method.
```
Comment (sims1253) [inline docs/Developers/Writing_Linter_Bears.rst]: So it is just a file I can give a bear right? Everything else is implemented in the bear?

Comment (NiklasMM) [inline docs/Developers/Writing_Linter_Bears.rst]: That's how I understand it, yes.

**Comment by sims1253** [comment]
ack ec1550c2619df9e680feb6d61962b1e44c94b362

**Comment by sims1253** [comment]
ack 9280aa5040549fc72d5b517f8b96912f3204bf70

**Comment by adtac** [comment]
@cherry-pick-bot

**Comment by adtac** [comment]
Merged here https://github.com/coala/coala/pull/3994

**Pull request closed without merging**